### PR TITLE
[webgpu] Add support for oneHot

### PR DESCRIPTION
**Pull request by haoyunfeix**
FEATURE

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5793)

### CODE REVIEW

**Comment by haoyunfeix** [comment]
@qjia7 @xhcao @axinging @gyagp PTAL

File: tfjs-backend-webgpu/src/kernels/onehot_webgpu.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import {WebGPUProgram} from './webgpu_program';
+
+export class OneHotProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['indices'];
+  uniforms = `onValue : f32; offValue : f32;`;
+  workGroupSize: [number, number, number] = [256, 1, 1];
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/onehot_webgpu.ts:30]: nit: currently, for non shared memory kernel, we usually use 64 instead 256 for code consistency.

**Review by qjia7 — APPROVED** [review]
LGTM with one nit.

**Review by qjia7 — APPROVED** [review]
LGTM

cc @lina128 @pyu10055

**Comment by qjia7** [comment]
@lina128 @pyu10055 @mattsoulanille Please help merge it, thanks.

**Comment by haoyunfeix** [comment]
merged in another PR, close this one.

**Pull request closed without merging**